Sources/MessagePack/Unpack.swift

```diff
@@ -1,12 +1,112 @@
+
 import Foundation
 
+
+/// Handle the remaining data without duplicating
+final class RemainderData {
+  let data:Data
+  private let bytes:UnsafePointer<UInt8>
+  private let rawBytes:UnsafeRawPointer
+  
+  /// Current location of where we are processing
+  private var offset:Int = 0 {
+    didSet {
+      isEmpty = offset >= data.endIndex
+    }
+  }
+  
+  /// Initialize with the original data source
+  ///
+  /// - Parameter data: The data to be unpacked
+  init(_ data:Data) {
+    self.data = data
+    rawBytes = (self.data as NSData).bytes
```

Comment:
Accessing NSData.bytes didn't appear to balloon the data, but it wasn't linux compatible anyways.

In order to accomplish the change I needed to change the return signatures of one of the public functions to return RemainderData instead of Data. As well, getting data objects from a message pack  does return copies of the data.